stumpy/mstump.py

```diff
@@ -315,9 +320,9 @@ def subspace(
         bins = _inverse_norm(n_bit)
         discretize_func = partial(_discretize, bins=bins)
 
-    subseqs, _, _ = core.preprocess(T[:, subseq_idx : subseq_idx + m], m)
+    subseqs, _, _, _ = core.preprocess(T[:, subseq_idx : subseq_idx + m], m)
```

Comment:
Here we just clean the `subseqs` by replacing its `nan` and `inf` with `0`. So, can we write a function just for that cleaning process and use it here? We are unnecessarily computing sliding mean, rolling stddev, and rolling isconstant.

Reply:
If you must, we could change this to something like:

```
subseqs = T[:, subseq_idx : subseq_idx + m].copy()
subseqs[~np.isfinite(subseqs)] = 0.0
```

Would that work?

Reply:
> However, there is no actual "rolling" mean/stddev/isconstant as this is only for a single subsequence.

Oh... I did not pay attention to this. If that is the case, then we can keep it as is since I think the existing code is more readable. (the only downside is that if we develop this code later to consider more-than-one subsequence, then the existing code might become a little bit slow). 

@seanlaw 
What do you prefer here?

Reply:
> the only downside is that if we develop this code later to consider more-than-one subsequence, then the existing code might become a little bit slow

I highly doubt that we'll add more subsequences in the future. The `subspace` function requires `subseq_idx` as its input, which forces a single subsequence and never more. Let's just keep it the same/unchanged